labs/content/labs/05-forms.md

```diff
@@ -244,47 +221,56 @@ render() {
 
 ```jsx
 handleSave = (values) => {
-  const { onCreate, onUpdate, history } = this.props;
+    const { history } = this.props;
 
-  const result = values._id ? onUpdate(values) : onCreate(values);
-  result.then(() => {
-    history.push('/employees');
-  });
+    const result = values._id ? this.onUpdate(values) : this.onCreate(values);
+    result.then(() => {
+      history.push('/employees');
+    });
 };
 ```
 
 * When handleSave is called we'll:
   - Check to see if the record being saved already has an ID - if yes it's an update, if not it's a create.
-  - We call the corresponding handler function to begin the appropriate async action method
-  - Once complete, we'll tell react-router to send the user back to the "/employees" route so they see the Table.
+  - We call the corresponding handler function to begin the appropriate async function
+  - Once complete, we'll tell react-router to send the user back to the "/employees" route so they see the table.
+
+* Finally, we'll implement the `onUpdate` and `onCreate` functions, to persist the employee with our PUT and POST APIs.
+
+```jsx
+ onUpdate = async employee => {
+    const response = await Axios.put(url(employee._id), employee);
+    return response.data;
+  };
+
+  onCreate = async employee => {
+    const response = await Axios.post(url(employee._id), employee);
+    return response.data;
+  };
+```
 
 &nbsp;
 
 ## Test the Employee Detail Component
 
 * Open **/src/employees/EmployeeDetail.test.js**
+* Use jest.mock to mock our API calls
 * Add a test to verify the component renders as expected:
 
 ```jsx
-let wrapper;
-
-beforeEach(() => {
-  const mockStore = configureStore();
-  wrapper = mount(
-    <Provider store={mockStore}>
-      <MemoryRouter>
-        <EmployeeDetail
-          match={{
-            params: {}
-          }}
-        />
-      </MemoryRouter>
-    </Provider>
-  );
-});
-
-it('should instantiate the Employee Detail Component', () => {
-  expect(wrapper.find(EmployeeDetail)).toIncludeText('Employee Detail');
+jest.mock('axios', () => ({
```

Comment:
Other unit tests are using moxios, but I prefer using pure Jest mocks. Maybe something to standardize in the future?